File: leapfrog-py/scripts/create_model_output.py
#!/usr/bin/env python3

"""Run leapfrog model and save output to specified dir
Usage:
  run_model <output-dir>

Arguments:
  <output-dir>  Path to save output to.

Options:
  -h --help                  Show this screen.
"""

from docopt import docopt
import os

from leapfrog_py import read_h5_file, run_model, save_h5_file

if __name__ == '__main__':
    args = docopt(__doc__)
    output_dir = args["<output-dir>"]
    
    if not os.path.exists(output_dir):
        os.mkdir(output_dir)


    parameters = read_h5_file("../r-package/tests/testthat/testdata/adult_parms_full.h5")
    ret = run_model(parameters)

    save_h5_file(ret, os.path.join(output_dir, "py-output.h5"))

File: leapfrogr/scripts/create_model_output.R
#!/usr/bin/env Rscript

usage <- "Run leapfrog model and save output to specified dir
Usage:
  run_model <output-dir>

Arguments:
  <output-dir>  Path to save output to.

Options:
  -h --help                  Show this screen.
"

dat <- docopt::docopt(usage)
names(dat) <- gsub("-", "_", names(dat), fixed = TRUE)

if (!dir.exists(dat$output_dir)) {
  dir.create(dat$output_dir, recursive = TRUE)
}

parameters <- leapfrog::read_parameters(testthat::test_path("testdata/adult_parms_full.h5"))
out <- leapfrog::run_model(parameters)

leapfrog:::save_hdf5_file(out, file.path(dat$output_dir, "r-output.h5"))
